src/hooks/web3/index.tsx

```diff
@@ -72,18 +72,26 @@ export function Provider({
     [web3ReactConnectors]
   )
 
-  // Attempt to connect eagerly if there is no user-provided provider.
+  const shouldEagerlyConnect = provider === undefined
   useEffect(() => {
     // Ignore any errors during connection so they do not propagate to the widget.
     if (connectors.user) {
       connectors.user.activate().catch(() => undefined)
       return
-    } else if (connectors.user !== null) {
+    } else if (shouldEagerlyConnect) {
       const eagerConnectors = [connectors.metaMask, connectors.walletConnect]
       eagerConnectors.forEach((connector) => connector.connectEagerly().catch(() => undefined))
     }
     connectors.network.activate().catch(() => undefined)
-  }, [connectors.metaMask, connectors.network, connectors.user, connectors.walletConnect])
+  }, [
+    connectors.metaMask,
+    connectors.network,
+    connectors.user,
+    connectors.walletConnect,
+    provider,
+    shouldEagerlyConnect,
```

Comment:
this will always change when provider changes right?

Reply:
yes. provider should be omitted.